### PR TITLE
aws - memorydb - add support for user and acl

### PR DESCRIPTION
**Pull request by jerryhxu**
Closes issue https://github.com/cloud-custodian/cloud-custodian/issues/9712

### CODE REVIEW

File: c7n/resources/memorydb.py [hunk 1]
```diff
@@ -67,8 +67,50 @@ class resource_type(TypeInfo):
...
+
+    https://docs.aws.amazon.com/memorydb/latest/devguide/what-is-memorydb-for-redis.html
+    """
+
+    class resource_type(TypeInfo):
+
+        service = 'memorydb'
+        enum_spec = ('describe_users', 'Users', None)
+        arn = 'ARN'
+        arn_type = 'cluster'
```
Comment (kapilt) [inline c7n/resources/memorydb.py:82]: bad arn type

File: c7n/resources/memorydb.py [hunk 2]
```diff
@@ -67,8 +67,50 @@ class resource_type(TypeInfo):
...
+
+    https://docs.aws.amazon.com/memorydb/latest/devguide/what-is-memorydb-for-redis.html
+    """
+
+    class resource_type(TypeInfo):
+
+        service = 'memorydb'
+        enum_spec = ('describe_acls', 'ACLs', None)
+        arn = 'ARN'
+        arn_type = 'cluster'
```
Comment (kapilt) [inline c7n/resources/memorydb.py:102]: bad arn type

File: c7n/resources/memorydb.py (hunk 1 above)
Comment (jerryhxu) [reply c7n/resources/memorydb.py]: Thanks for pointing out. I fixed it.

File: c7n/resources/memorydb.py (hunk 2 above)
Comment (jerryhxu) [reply c7n/resources/memorydb.py]: Thanks for pointing it out. I fixed it.

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**